pkg/services/ngalert/models/alert_rule.go

```diff
@@ -26,6 +26,13 @@ var (
 	ErrAlertRuleFailedValidation          = errors.New("invalid alert rule")
 	ErrAlertRuleUniqueConstraintViolation = errors.New("a conflicting alert rule is found: rule title under the same organisation and folder should be unique")
 	ErrQuotaReached                       = errors.New("quota has been exceeded")
+	// ErrNoDashboard is returned when the alert rule does not have a Dashboard UID
+	// in its annotations or the dashboard does not exist.
+	ErrNoDashboard = errors.New("no dashboard")
```

Comment:
Moved from `image` as these errors can make sense in many places. I alternatively considered deleting these, and using `dashboards.DashboardNotFound` and `dashboards.DashboardPanelNotFound` straight up. Does anyone have a strong preference?

Reply:
I don't have a strong preference; at the moment, models it's just a hodgepodge for everything - at least if we keep them here, we only have 1 place to move _from_ eventually.